cookbook/bundles/tests.rst

```diff
@@ -0,0 +1,311 @@
+
+.. code-block:: bash
+
+    $ composer require --dev phpunit/phpunit
+
+.. note::
+
+    The steps should be similar with other tests frameworks.
+
+Then the second one is to configure it to use Composer's autoloading:
+
+.. configuration-block::
+
+    .. code-block:: xml
+
+        <?xml version="1.0" encoding="UTF-8"?>
+        <!-- phpunit.xml.dist -->
+
+        <!-- http://phpunit.de/manual/4.3/en/appendixes.configuration.html -->
+        <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/4.3/phpunit.xsd"
+            backupGlobals="false"
+            colors="true"
+            bootstrap="./vendor/autoload.php"
+        >
+            <testsuites>
+                <testsuite name="Test Suite">
+                    <directory>./Tests/</directory>
+                </testsuite>
+            </testsuites>
+
+            <filter>
+                <whitelist>
+                    <directory>.</directory>
+                    <exclude>
+                        <directory>./Resources</directory>
+                        <directory>./Tests</directory>
+                    </exclude>
+                </whitelist>
+            </filter>
+        </phpunit>
+
+With these two simple steps it becomes possible to run the test suite with the following command:
+
+.. code-block:: bash
+
+    $ vendor/bin/phpunit
+
+Functional Web Tests
+~~~~~~~~~~~~~~~~~~~~
+
+As advised in the official best practices (smoke testing), writing tests for
+controllers can be done by simply checking the status code::
+
+    // Tests/Controller/DemoControllerTest.php
+    namespace Acme\MyBundle\Tests\Controller;
+
+    use Acme\MyBundle\Tests\AppKernel;
+    use Symfony\Component\HttpFoundation\Request;
+
+    class DemoControllerTest extends \PHPUnit_Framework_TestCase
+    {
+        private $app;
+
+        protected function setUp()
+        {
+            $this->app = new AppKernel('test', false);
+            $this->app->boot();
+        }
+
+        public function testItRunsSuccessfully()
+        {
+            $headers = array('CONTENT_TYPE' => 'application/json');
+            $content = array('parameter' => 'value');
+            $request = Request::create('/demo', 'POST', array(), array(), array(), $headers, $content);
+
+            $response = $this->app->handle($request);
+
+            $this->assertSame(200, $response->getStatusCode(), $response->getContent());
+        }
+    }
+
+Functional CLI Tests
+~~~~~~~~~~~~~~~~~~~~
+
+As advised in the official best practices (smoke testing), writing tests for
+commands can be done by simply checking the exit code::
+
+    // Tests/Command/DemoCommandTest.php
+    namespace Acme\MyBundle\Tests\Command;
+
+    use Acme\MyBundle\Tests\AppKernel;
+    use Symfony\Bundle\FrameworkBundle\Console\Application;
+    use Symfony\Component\Console\Tester\ApplicationTester;
+
+    class DemoCommandTest extends \PHPUnit_Framework_TestCase
```

Comment:
I know, but in this precise situation it's not really necessary since we're already loading the Kernel.